ledger/tests/ledger.rs

```diff
@@ -81,3 +81,56 @@ async fn user_buys_and_sells_usd() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test]
+async fn adjust_exchange_position() -> anyhow::Result<()> {
+    let pool = init_pool().await?;
+    let ledger = Ledger::init(&pool).await?;
+
+    let initial_exchange_balance = ledger
+        .balances()
+        .exchange_position_account_balance(OKEX_POSITION_ID)
+        .await?
+        .map(|b| b.settled())
+        .unwrap_or(Decimal::ZERO);
+
+    ledger
+        .adjust_exchange_position(
+            pool.begin().await?,
+            dec!(-10000),
+            OKEX_POSITION_ID,
+            "okex".to_string(),
+            "BTC-USD-SWAP".to_string(),
+        )
+        .await?;
+    let balance_after_first_adjustment = ledger
+        .balances()
+        .exchange_position_account_balance(OKEX_POSITION_ID)
+        .await?
+        .unwrap()
+        .settled();
+    assert_eq!(
+        balance_after_first_adjustment - initial_exchange_balance,
+        dec!(100)
+    );
+    ledger
+        .adjust_exchange_position(
+            pool.begin().await?,
+            dec!(-9000),
+            OKEX_POSITION_ID,
+            "okex".to_string(),
+            "BTC-USD-SWAP".to_string(),
+        )
+        .await?;
+    let balance_after_second_adjustment = ledger
+        .balances()
+        .exchange_position_account_balance(OKEX_POSITION_ID)
+        .await?
+        .unwrap()
+        .settled();
+    assert_eq!(
+        balance_after_second_adjustment - balance_after_first_adjustment,
```

Comment:
I don't understand this assert. You are executing the same template twice and only asserting the effect of the second application. Wouldn't it be better to apply each template once and assert `balance_after_second_adjustment - initial_balance` - that way both templates must have worked.